### ISSUE TITLE
NaN loss with SDPA attention when local_batch_size > 1 on Gemma 4 4B

### ISSUE THREAD

**Issue by HuiyingLi** [NVIDIA-NeMo/Automodel#1883]
## Summary

Training `gemma4_4b.yaml` with `attn_implementation=sdpa` and `local_batch_size=2` produces NaN loss early in training.

This is a known limitation in transformers, opening this issue to help with tracking.
- https://github.com/huggingface/transformers/issues/45201

## Environment

- Model: `google/gemma-4-E4B-it`
- Config: `examples/vlm_finetune/gemma4/gemma4_4b.yaml`
- `transformers`: 5.5.0.dev0
- `torch`: 2.11.0a0+eb65b36914.nv26.02 (CUDA 13.1)
- Hardware: 8× H100 80GB, single node
- Precision: bf16, FSDP2

## Repro

```bash
torchrun --nproc-per-node=8 nemo_automodel/recipes/vlm/finetune.py \
  -c examples/vlm_finetune/gemma4/gemma4_4b.yaml \
  --step_scheduler.max_steps=30 \
  --step_scheduler.local_batch_size=2 \
  --step_scheduler.global_batch_size=16 \
  --model.attn_implementation=sdpa
```

## Observed

`grad_norm` becomes NaN at step 11, loss NaN from step 12 onward:

```
step 10 | loss 2.6030 | grad_norm  9.9375
step 11 | loss 2.3224 | grad_norm     nan
step 12 | loss    nan | grad_norm     nan
...
[val] step 29 | loss nan
```

wandb: https://wandb.ai/Nemo-automodel/gemma4-4b-attn-lbs/runs/4yb1b74k

<img width="934" height="240" alt="Image" src="https://github.com/user-attachments/assets/08fa8b83-1fe7-468c-a49b-601f13f69b72" />

**Comment by khazic** [comment]
I previously ran into the same issue when using your scripts, and I already fixed that bug. However, my environment uses PyTorch 2.10 and Transformers 5.5.3, so I’ll test it in your environment to see exactly where the problem is.

**Comment by athitten** [comment]
Note to avoid confusion: This issue is orthogonal to the [lack of FA support for gemma4](https://github.com/huggingface/transformers/issues/45201). They are two independent issues.
cc: @akoumpa @HuiyingLi

**Comment by athitten** [comment]
Can't be reproduced with transformers v5.5 version. Closing for now, to open a new issue if we encounter NaN again.
